docs/upgrade.md

```diff
@@ -161,6 +161,27 @@ Synapse will log a warning when a module uses the deprecated method, to help
 administrators find modules using it.
 
 
+## Package extra `url_preview` now called `url-preview`
+
+This is compliant with [PEP 685](https://peps.python.org/pep-0685/#specification) and is necessary to use the URL preview option when Synapse is built with poetry-core 1.3.0 or greater. [#14079](https://github.com/matrix-org/synapse/issues/14079) has some more context.
+
+If you are installing Synapse from source with this extra specifically named, e.g. with
+
+```
+pip install /path/to/synapse[url_preview]
+```
+
+or
+
+```
+poetry install --extras url_preview
+```
```

Comment:
right, fair.

> I think installations that specifically name url_previews are unlikely.

Yeah. On this occasion, I feel like this is unlikely to be useful to anyone. We do need to find some sort of balance in these upgrade notes: if they describe every single change, then people won't read them at all. IMHO this is on the "not worth including" side of the balance (though it might be worth an explicit shoutout in the changelog). YMMV though.

Reply:
I mainly want to have something here to point people to if they encounter this problem on upgrade. Can remove it if you think it's too niche.

Reply:
Ahh, sorry. I missed your earlier comment.

I'll omit this from the notes and add a "BTW" in the changelog.

Reply:
This is a722cf52b.